modules/fedimint-ln-client/src/receive.rs

```diff
@@ -202,12 +195,32 @@ impl LightningReceiveConfirmedInvoice {
         loop {
             match global_context
                 .module_api()
-                .wait_preimage_decrypted(contract_id)
+                .get_decrypted_preimage_status(contract_id)
                 .await
             {
-                Ok((incoming_contract_account, preimage)) => match preimage {
-                    Some(_) => return Ok(incoming_contract_account),
-                    None => return Err(LightningReceiveError::InvalidPreimage),
+                Ok((incoming_contract_account, status)) => match status {
+                    DecryptedPreimageStatus::Pending => {
+                        // only when we are sure that the invoice is still pending that we can check
+                        // for a timeout
+                        let now = fedimint_core::time::now();
+                        match invoice.timestamp().checked_add(invoice.expiry_time()) {
```

Comment:
[`timestamp()` is only available with `std`](https://docs.rs/lightning-invoice/latest/lightning_invoice/struct.Bolt11Invoice.html#method.timestamp), this might break MutinyWallet again. @benthecarman @TonyGiorgio There are alternative, more clunky methods to get the timestamp, e.g. [`duration_since_epoch`](https://docs.rs/lightning-invoice/latest/lightning_invoice/struct.Bolt11Invoice.html#method.duration_since_epoch)

Reply:
Yeah that will break

Reply:
Yeah if you can help it, `duration_since_epoch` is better, though in this case we could alter our fork to allow that call, since I followed the chain and it doesn't appear to be calling `SystemTime::now()`.